app/wait_time/api.py

```diff
@@ -0,0 +1,40 @@
+from flask import jsonify
+from app.auth import with_auth_user
+from app.user.user import User
+from typing import Dict, Any
+from .location import UserLocation
+from .service import uid_to_aid, update_location, add_wait_time_suggestion
+from app.common import BaseApiError
+
+
+@with_auth_user
+def update_user_location(
+    user: User, location_data: Dict[str, float], token_info: Dict[str, Any]
+):
+    try:
+        user_location = UserLocation.from_dict(
+            uid_to_aid(token_info["uid"]), location_data
+        )
+        update_location(user_location)
+    except BaseApiError as e:
+        return e.build_error()
+    except Exception as e:
+        return {"error": str(e)}, 500
+    return None, 204
+
+
+@with_auth_user
+def submit_user_estimate(
+    user: User, estimate_data: Dict[str, int], poi_id: str, **kwargs
+):
+    # TODO: implement user estimate submission
```

Comment:
Apparently you can specify more restrictions on data in the OpenAPI spec beyond just required parameters. For ints, you can specify minimum and maximum values, and you can also specify enums. This may be enough for now but if we find that we need something with more functionality in the future we can use this

Reply:
The openapi stuff is definitely not going to be enough I think. We don't have to include it in this PR but if you could make a ticket to address it that would be 💯

Reply:
Done. #44